doc/syslog/syslog-rate-limit-design.md

```diff
@@ -0,0 +1,376 @@
+$SystemLogRateLimitBurst 20000
+```
+There is no rate limiting configured on host side for now.
+
+The SystemLogRateLimitInterval determines the amount of time that is being measured for rate limiting. The SystemLogRateLimitBurst defines the amount of messages, that have to occur in the time limit of SystemLogRateLimitInterval, to trigger rate limiting. For example, SystemLogRateLimitInterval=300, SystemLogRateLimitBurst=20000, it means that if one daemon generate more than 20000 messages in 300 seconds, rsyslogd will start to drop messages after that(FIFO).
+
+This feature allows user to configure SystemLogRateLimitInterval and SystemLogRateLimitBurst for host, containers.
+
+### Requirements
+
+- Support syslog message rate limit configuration for host and containers
+- New CLI for rate limit configuration
+- Have rate limit configuration persistent
+- Default rate limit shall be applied if no configuration provided
+- Both multi ASIC and single ASIC platform shall use rsyslog-container.conf.j2 as template
+- APP extension shall be able to declare its capability of this feature
+- APP extension is responsible for reading/listening rate limit configuration from CONFIG DB and set rsyslog configuration accordingly.
+
+### Architecture Design
+
+rsyslog.conf.j2 and rsyslog-container.conf.j2 shall be extended to accept template variable for SystemLogRateLimitInterval and SystemLogRateLimitBurst. rsyslog.conf for single ASIC platform will be removed and replaced by rsyslog-container.conf.j2.
+
+Host side:
+
+![host_config_on_fly](/doc/syslog/images/host_rate_limit_config_flow.svg).
+
+Change syslog message rate limit while container is running:
+
+![container_config_on_fly](/doc/syslog/images/container_rate_limit_config_flow.svg).
+
+For app extension, CLI shall only put the configuration to CONFIG DB and let app extension itself to handle it. How app extension handle CONFIG DB change is out of the feature scope.
+
+Load syslog message rate limit configuration during startup is already covered by existing flow, no change of the flow is required:
+
+- Host: rsyslog-config service automatically loads rate limit configuration from CONFIG DB and apply it to rsyslog
+- Container: each container have a startup script which automatically generates rsyslog configuration and apply it to rsyslog in container
+
+> Note: according to test, syslog rate limit configuration on host side would not affect container side.
+
+### High-Level Design
+
+Changes shall be made into sonic-utilities, sonic-buildimage. CLI changes of sonic-utilities will be covered by chapter "Configuration and management".
+
+> Note: Code present in this design document is only for demonstrating the design idea, it is not production code.
+
+#### init_cfg.json.j2
+New tables shall be added to CONFIG DB to store the rate limit configuration. init_cfg.json.j2 shall be extended to define the default value for each built-in containers.
+
+```
+...
+    "SYSLOG_CONFIG": {
+        "GLOBAL": {
```

Comment:
GLOBAL seems to indicate that the settings are applicable for Host and Containers, suggest to have to better key name.

Reply:
GLOBAL is a typical way to indicate that the entry is for host side. Please refer to https://github.com/sonic-net/sonic-buildimage/blob/53a281008a96edcc22d0536ca4357a18f071c04e/src/sonic-yang-models/yang-models/sonic-auto_techsupport.yang#L31

Reply:
Ok, please a comment/description in the YANG for clarity, thanks!